src/core/qgsfeaturerequest.cpp

```diff
@@ -485,6 +485,23 @@ QSet<QString> QgsFeatureRequest::OrderBy::usedAttributes() const
   return usedAttributes;
 }
 
+QSet<int> QgsFeatureRequest::OrderBy::usedAttributeIndices( const QgsFields &fields ) const
+{
+  QSet<int> usedAttributeIdx;
+  for ( const OrderByClause &clause : *this )
+  {
+    for ( const QString &fieldName : clause.expression().referencedColumns() )
```

Comment:
A const copy needs to be taken to avoid a detach of containers with range-based-for loops.
```suggestion
    const auto referencedColumns = clause.expression().referencedColumns();
    for ( const QString &fieldName : referencedColumns )
```